### ISSUE TITLE
REPL crash when cancelling entr

### ISSUE THREAD

**Issue by rikhuijzer** [timholy/Revise.jl#602]
Pressing CTRL + C after running `entr` crashes the REPL completely. 

First part of the stacktrace:
```
ERROR: TaskFailedException
Stacktrace:
  [1] try_yieldto(undo::typeof(Base.ensure_rescheduled))
    @ Base ./task.jl:705
  [2] wait
    @ ./task.jl:764 [inlined]
  [3] wait(c::Base.GenericCondition{ReentrantLock})
    @ Base ./condition.jl:106
  [4] take_buffered(c::Channel{Any})
    @ Base ./channels.jl:389
  [5] take!
    @ ./channels.jl:383 [inlined]
  [6] repl_backend_loop(backend::REPL.REPLBackend)
    @ REPL /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:195
  [7] start_repl_backend(backend::REPL.REPLBackend, consumer::Any)
    @ REPL /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:185
  [8] run_repl(repl::REPL.AbstractREPL, consumer::Any; backend_on_current_task::Bool)
    @ REPL /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:317
  [9] run_repl(repl::REPL.AbstractREPL, consumer::Any)
    @ REPL /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:305
 [10] (::Base.var"#878#880"{Bool, Bool, Bool})(REPL::Module)
    @ Base ./client.jl:387
 [11] #invokelatest#2
    @ ./essentials.jl:707 [inlined]
 [12] invokelatest
    @ ./essentials.jl:706 [inlined]
 [13] run_main_repl(interactive::Bool, quiet::Bool, banner::Bool, history_file::Bool, color_set::Bool)
    @ Base ./client.jl:372
 [14] exec_options(opts::Base.JLOptions)
    @ Base ./client.jl:302
 [15] _start()
    @ Base ./client.jl:485

    nested task error: IOError: stream is closed or unusable
    Stacktrace:
     [1] check_open(x::Base.TTY)
       @ Base ./stream.jl:386
     [2] raw!(t::REPL.Terminals.TTYTerminal, raw::Bool)
       @ REPL.Terminals /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/Terminals.jl:138
     [3] prompt!(term::REPL.Terminals.TextTerminal, prompt::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
       @ REPL.LineEdit /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/LineEdit.jl:2557
     [4] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
       @ REPL.LineEdit /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/LineEdit.jl:2436
     [5] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
       @ REPL /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/REPL/src/REPL.jl:1126
     [6] (::REPL.var"#44#49"{REPL.LineEditREPL, REPL.REPLBackendRef})()
       @ REPL ./task.jl:406
...
```

I think this is easy to reproduce because it used to be like this on Julia 1.5 as well if I remember correctly. Let me know if you can't reproduce it and feel free to close this issue if you don't see this as a priority.

```
julia> versioninfo()
Julia Version 1.6.0-rc1
Commit a58bdd9010 (2021-02-06 15:49 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
Environment:
  JULIA_NUM_THREADS = 4
```

**Comment by timholy** [comment]
`entr` is largely community-maintained so a volunteer would be great!

**Comment by ffevotte** [comment]
> I think this is easy to reproduce because it used to be like this on Julia 1.5 as well if I remember correctly. Let me know if you can't reproduce it

Not sure I completely understood your problem, but in my early attempts I did not manage to reproduce your issue:
```
julia> versioninfo()
Julia Version 1.6.0-rc1
Commit a58bdd9010 (2021-02-06 15:49 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i5-9300H CPU @ 2.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
Environment:
  JULIA_PROJECT = @.

julia> using Revise

julia> entr(["/tmp/"]) do
         println("update")
       end
update
update
update
update
^C
julia> println("OK")
OK
```

The same happens on my system with Julia 1.5.3 and nightly.

Could you please give us more details regarding your system and the conditions when this happens?

**Comment by rikhuijzer** [comment]
Hmm, yes, indeed. I just tested it in Docker and also cannot reproduce the error there. I've cleaned out `~/.julia` and the issue then still occurs on NixOS. Therefore, it is then most likely related to my NixOS setup (https://discourse.julialang.org/t/build-julia-on-nixos/35129/43). Now that I think of it, more packages crash where they shouldn't, so this is probably an NixOS problem. I'll close this issue. Thanks for looking into it.
